README.md

```diff
@@ -20,127 +19,273 @@ let thirdNumber = numbers[2]
 ```swift
  if let index = names.indexOf("Jim") {
     names.removeAtIndex(index)
  }
- 
+
  // ["Sarah", "Bran", "Sansa"]
 ```
 
-* Referring to `count`, they should be exposed to the following where it is explained thoroughly what is going on: 
+- Referring to `count`, they should be exposed to the following where it is explained thoroughly what is going on:
 
 ```swift
 let rainbowColors = ["Red", "Orange", "Yellow", "Green", "Blue", "Indigo", "Violet"]
- 
+
  let numberOfColors = rainbowColors.count
  // 7
- 
+
  print("There are \(numberOfColors) colors in the rainbow.")
  // prints "There are 7 colors in the rainbow.
- 
+
  if rainbowColors.count > 5 {
     print("There are more than 5 colors in a rainbow.")
  }
  //prints "There are more than 5 colors in a rainbow."
 ```
 
-* Referring to `isEmpty` - I like the idea of stepping them through the following two examples:
+- Referring to `isEmpty` - I like the idea of stepping them through the following two examples:
 
 ![deliLine](http://i.imgur.com/ivgaSon.png?1)
+
 ```swift
  var deliLine = [
     "Albert Einstein",
     "Isaac Newton",
     "Galileo Galilei",
     "Marie Curie"
  ]
- 
+
  if deliLine.isEmpty {
     print("The deli line is empty")
  } else {
     print("There are people waiting to eat!")
  }
- 
+
  // prints "There are people waiting to eat!"
 ```
+
 ![Moons](http://i.imgur.com/reNsnn6.png?1)
+
 ```swift
  let moonsOfJupiter = [
     "Europa",
     "Ganymedge",
     "Io",
     "Callisto"
 ]
- 
+
  if !moonsOfJupiter.isEmpty {
     for moon in moonsOfJupiter {
         print("\(moon) is a moon orbiting Jupiter.")
     }
  }
- 
+
  // Europa is a moon orbiting Jupiter.
  // Ganymede is a moon orbiting Jupiter.
  // Io is a moon orbiting Jupiter.
  // Callisto is a moon orbiting Jupiter.
 ```
 
+--------------------------------------------------------------------------------
+
+In the last lesson you briefly saw a 'method', which can be thought of as a pre-written function that helps you accomplish common tasks. They are contextual, in that, for example, a method that works on an array may not work for a String.
+
+The method you saw in the last example was `.append()` that adds a value to the end of an array. You could achieve this in several ways which would involve far more code and calculation, I think you'll agree that `.append()` is far more convenient. There are many other methods for arrays, and in this lesson you'll look at some others.
+
+## Enumeration
+
+Before going any further, there's one other array concept to introduce.
+
+Say you have a list of instruction steps, if steps need to be added, removed, or changed at a particular index in the array, then you need to know what the index of the value is. When you have a simple list of unchanging values, then maybe this is simple, but with a larger list processed in a loop, there is another way, you can use the `.enumerate` method to have not only the value of the current loop index, but the index position itself.
+
```

Comment:
I love this. I think we can word this a little different.

Say you have an array of string literals which represent a list of instructions to create great tea. If we want to print these instructions to our user, we can do so like this if we like:

```swift
print("1. \(toMakeTea[0])")
print("2. \(toMakeTea[1])")
// and so on

// prints "1. Boil Water"
// prints "2. Add tea bag to cup"
```

This way of accomplishing this task is incredibly tedious. Knowing how a `for-in` loop works, we can iterate over each value in our array of strings. But, we're also in need of the index because we want to display to the user numbered instructions. This is possible using the `enumerate()` method. Similar to how you call `uppercaseString` on the your string variable, you call `enumerate()` on our array variable. We iterate over that `enumerate()` method call on our variable to be able to get not only the elements within that array, but the index of each element along side it.

```swift
for (index, step) in toMakeTea.enumerate() {
    print("\(index + 1). \(step)")
}

// 1. Boil Water
// 2. Add tea bag to cup
// 1. Boil Water
// 2. Add tea bag to cup
// 3. Wait ten minutes
// 4. Add Milk
// 5. Drink
```